src/main/java/fi/aalto/cs/apluscourses/utils/CollectionUtil.java

```diff
@@ -37,11 +37,24 @@ private CollectionUtil() {
   }
 
   /**
-   *  Returns the index of the given element in an iterator.
+   * Like mapWithIndex, but reverse.
+   */
+  public static <T, R> List<R> mapWithIndexReverse(@NotNull List<T> list,
+                                                   @NotNull BiFunction<T, Integer, R> func,
+                                                   int startIndex) {
+    int n = list.size();
+    return IntStream
+        .range(0, n)
+        .mapToObj(i -> func.apply(list.get(n - i - 1), n - (startIndex + i) + 1))
+        .collect(Collectors.toList());
+  }
```

Comment:
Works nicely, but now the `mapWithIndex` is unused (used only in its own test)
Either:
- remove `mapWithIndex` and change the corresponding test to test `mapWithIndexReverse`
- keep `mapWithIndex` and reverse the list inside ExerciseViewModel.

Reply:
So does this method work like (with list `{a,b,c,d}` and `startIndex = 1`):
```
(4,d)
(3,c)
(2,b)
(1,a)
```
In that case, I think it's better to use `mapWithIndex` and then reverse. It's much clearer and less redundant in code. The performance aspect is not important in our case, I think.